src/apps/containers.clj

```diff
@@ -84,54 +72,40 @@
                                  (where {:id (uuidify tool-uuid)}))))]
     (image-info image-id :auth? auth?)))
 
-(defn- get-tag
-  [image-map]
-  (if-not (contains? image-map :tag)
-    "latest"
-    (:tag image-map)))
+(defn find-image-by-name-and-tag
+  "Finds the image matching the given name and tag in the container_images table."
+  [{:keys [name tag] :or {tag "latest"}}]
+  (first (select container-images (where {:name name
+                                          :tag  tag}))))
+
+(defn find-or-add-image-info
+  "Finds or inserts an image with the given name and tag in the container_images table."
+  [{:keys [name tag url] :or {tag "latest"} :as image-map}]
+  (let [existing-image (find-image-by-name-and-tag image-map)]
```

Comment:
I was thinking about cleaning this function up.
I like your suggestion, so I'll refactor this before I merge this PR 👍